File: zjctools/src/main/java/com/android/zjctools/glide/ZGlideModule.java
package com.android.zjctools.glide;


import com.bumptech.glide.annotation.GlideModule;
import com.bumptech.glide.module.AppGlideModule;

@GlideModule
public class ZGlideModule extends AppGlideModule {
    /**
     * 如需要设置图片缓存路径和缓存大小，则复制释放下面的方法即可，否则不需要重写
     */
//    @Override
//    public void applyOptions(Context context, GlideBuilder builder) {
//        //        super.applyOptions(context, builder);
//        int size = 200 * 1024 * 1024;
//        String dir = ZConstants.CACHE_IMAGES;
//        builder.setDiskCache(new ExternalPreferredCacheDiskCacheFactory(context, dir, size));
//    }
//
//    @Override
//    public void registerComponents(Context context, Glide glide, Registry registry) {
//        super.registerComponents(context, glide, registry);
//    }
//
//    @Override
//    public boolean isManifestParsingEnabled() {
//        return super.isManifestParsingEnabled();
//    }
}

File: zjctools/src/main/java/com/android/zjctools/glide/ZGlideModule.kt
package com.android.zjctools.glide

import android.content.Context

import com.bumptech.glide.GlideBuilder
import com.bumptech.glide.annotation.GlideModule
import com.bumptech.glide.load.engine.cache.ExternalPreferredCacheDiskCacheFactory
import com.bumptech.glide.module.AppGlideModule



@GlideModule
class ZGlideModule : AppGlideModule() {
    override fun applyOptions(context: Context, builder: GlideBuilder) {
        //        super.applyOptions(context, builder);
        val size = 256 * 1024 * 1024
        val dir = "imgs"
        builder.setDiskCache(ExternalPreferredCacheDiskCacheFactory(context, dir, size.toLong()))
    }

}